customer-backend/src/main/java/io/eventuate/examples/tram/ordersandcustomers/customers/service/OrderEventConsumerEventHandlers.java

```diff
@@ -0,0 +1,70 @@
+package io.eventuate.examples.tram.ordersandcustomers.customers.service;
+
+import io.eventuate.examples.tram.ordersandcustomers.commondomain.CustomerCreditReservationFailedEvent;
+import io.eventuate.examples.tram.ordersandcustomers.commondomain.CustomerCreditReservedEvent;
+import io.eventuate.examples.tram.ordersandcustomers.commondomain.CustomerValidationFailedEvent;
+import io.eventuate.examples.tram.ordersandcustomers.commondomain.OrderCreatedEvent;
+import io.eventuate.examples.tram.ordersandcustomers.customers.domain.Customer;
+import io.eventuate.examples.tram.ordersandcustomers.customers.domain.CustomerCreditLimitExceededException;
+import io.eventuate.tram.events.publisher.DomainEventPublisher;
+import io.eventuate.tram.events.subscriber.DomainEventEnvelope;
+import io.micronaut.spring.tx.annotation.Transactional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import java.util.Collections;
+
+@Singleton
+public class OrderEventConsumerEventHandlers {
+  private Logger logger = LoggerFactory.getLogger(getClass());
+
+  @Inject
+  private DomainEventPublisher domainEventPublisher;
+
+  @PersistenceContext
+  private EntityManager entityManager;
+
+  @Transactional
```

Comment:
Event handlers should not be @Transactional- The transaction is managed by Eventuate Tram

Reply:
https://gist.github.com/dartartem/715f1dce08507674fdc321ce66a066ce
Entity manager requires transaction to work.
And I think more correctly is: eventuate tram transaction (common jdbc operations) is managed by application. I will write it in my report in more details.

Reply:
I think there are missing `@Beans`:

* DuplicateDetectingMessageHandlerDecorator which invokes SqlTableBasedDuplicateMessageDetector
* SqlTableBasedDuplicateMessageDetector which begins a transaction

Reply:
I will check, thank you